cerberus/validator.py

```diff
@@ -723,7 +723,7 @@ def __normalize_coerce(self, processor, field, value, nullable, error):
         try:
             return processor(value)
         except Exception as e:
-            if not nullable and e is not TypeError:
+            if not nullable or value is not None:
```

Comment:
i find this term more obvious: `not (nullable and value is None)`

Reply:
second thought: should the coercion get tried at all when the part in brackets is true?

Reply:
I changed the condition.

Not sure there's an obvious answer to whether the coercion should be tried or not. Somebody might want to coerce `None` to something else? Or would the validation in that case fail at a later stage anyway?

Reply:
I just realized that NOT trying the coercion for nullable fields when the value is None would fix this issue: https://github.com/pyeve/cerberus/issues/427

To be honest, I don't feel I'm in a position to make the right call here. So happy to make this change or not.

Reply:
thanks for digging this out. that issue concluded to postpone changes until the 2.0 release; and i think i shall document that this still needs to be layed out design-wise.